Fix schema_assets_filter configuration docs

ittmann opened the pull request:
The callable set as schemaAssetsFilter should return false for tables that you want filtered out 

See for example https://github.com/doctrine/dbal/blob/3.6.x/src/Configuration.php#L168 and https://github.com/doctrine/dbal/blob/4.0.x/src/Schema/AbstractSchemaManager.php#L204

ittmann committed:
Fix schema_assets_filter configuration docs

The callable set as schemaAssetsFilter should return false for tables that you want filtered out
See e.g. https://github.com/doctrine/dbal/blob/3.6.x/src/Configuration.php#L167

ittmann committed:
Fix syntax error, unexpected token ";"

SenseException commented:
A description about what to expect on true and false would be insightful. Something like "If the function returns true, then..."

driehle reviewed (changes requested):
Yes, please add something in the writting text as well on how this filter works!

DIReports commented:
My intent with this PR was to fix the documentation error.
Documentation should probably be added to the [DBAL Configuration documentation](https://www.doctrine-project.org/projects/doctrine-dbal/en/current/reference/configuration.html) and/or [Configuration class](https://github.com/doctrine/dbal/blob/3.5.x/src/Configuration.php#L44)

With the latest package versions (dbal 3.5.3 and migrations 3.5.5) adding the 'migrations' table to the asset filter breaks doctrine migrations (and not adding it makes the schema-tool when run with the the "--complete" option drop the table)

ittmann committed:
Update schema_assets_filter documentation

DIReports commented:
Adding a table to the schema_asset_filter will not prevent an entity table from being created, only tables not owned by the DBAL from being dropped, I changed this in the first line.

Also I added some explanation on how the schema_assets_filter works and removed the 'migrations' table from the example as adding it to the asset filter currently breaks doctrine migrations.

ittmann commented:
> With the latest package versions (dbal 3.5.3 and migrations 3.5.5) adding the 'migrations' table to the asset filter breaks doctrine migrations (and not adding it makes the schema-tool when run with the the "--complete" option drop the table)

Opened issue https://github.com/doctrine/migrations/issues/1317
**Adding the migrations table to doctrine (dbal) schema assets filter breaks migrations**

ittmann committed:
Fix Static Analysis / PHPStan

Fix Ignored error pattern #Parameter \#1 .* of method Doctrine\\Common\\DataFixtures\\Executor\\ORMExecutor::execute\(\)# in path
         tests/Paginator/AdapterTest.php was not matched in reported errors.

ittmann committed:
Update schema_assets_filter documentation

TomHAnderson commented on phpstan.neon:
```diff
@@ -15,9 +15,5 @@ parameters:
         -
             message: '#Parameter \#1 .* of method DoctrineORMModule\\Options\\Configuration.* stdClass given#'
             path: tests/Options/ConfigurationOptionsTest.php
-        -
```
This was removed in https://github.com/doctrine/DoctrineORMModule/commit/7f112e995ab42203ceff3dd8bd8c12cba4c6939d
